### PR TITLE
Changes keybinds to use signals

### LINKED ISSUE #1647: Keybinds runtime on non-human mobs
This is on master code, due to #1602 . A Runner, for example, does not have `quick_equip()`:
```
[00:40:09] Runtime in human.dm, line 14: undefined proc or verb /mob/living/carbon/xenomorph/runner/quick equip(). 
proc name: down (/datum/keybinding/human/quick_equip/down)
usr: Psykzz/(Young Runner (53))
usr.loc: (Medical Upper (96, 181, 3))
src: quick_equip (/datum/keybinding/human/quick_equip)
call stack:
quick_equip (/datum/keybinding/human/quick_equip): down(Psykzz (/client))
Psykzz (/client): keyDown("E")
```

This could be solved by sending a signal and making humans (and whoever else we want to) register it, or by using an intermediate redefined proc, instead of the current:
```
/datum/keybinding/human/quick_equip/down(client/user)
    var/mob/living/carbon/human/H = user.mob
    H.quick_equip()
    return TRUE
```

### PR DESCRIPTION
**Pull request by Rohesie**
Fixes #1647 
As described in there, in the signal solution.

### CODE REVIEW

File: code/datums/keybinding/human.dm [hunk 1]
```diff
@@ -9,9 +9,9 @@
     full_name = "Quick equip"
     description = ""
 
+
 /datum/keybinding/human/quick_equip/down(client/user)
-    var/mob/living/carbon/human/H = user.mob
-    H.quick_equip()
+    SEND_SIGNAL(user.mob, COMSIG_KB_QUICKEQUIP)
```
Comment (spookydonut) [inline code/datums/keybinding/human.dm]:
```suggestion
    if(SEND_SIGNAL(user.mob, COMSIG_KB_QUICKEQUIP) & comsig keybind handled or something) 
          Return true
```
Do the same for the others, only return on keybinds if it actually does the bind or you can't double up this key for xenos who don't need it

File: code/__DEFINES/components.dm [hunk 2]
```diff
@@ -190,3 +190,8 @@
 #define COMSIG_XENOABILITY_LUNGE "xenoability_lunge"
 #define COMSIG_XENOABILITY_FLING "xenoability_fling"
 #define COMSIG_XENOABILITY_PUNCH "xenoability_punch"
+
+// human signals for keybindings
+#define COMSIG_KB_QUICKEQUIP "keybinding_quickequip"
+#define COMSIG_KB_HOLSTER "keybinding_holster"
+#define COMSIG_KB_UNIQUEACTION "keybinding_uniqueaction"
```
Comment (spookydonut) [inline code/__DEFINES/components.dm]:
```suggestion
#define COMSIG_KB_UNIQUEACTION "keybinding_uniqueaction"
	#define COMSIG_KB_ACTIVATED 1

```

File: code/datums/keybinding/human.dm [hunk 3]
```diff
@@ -1,39 +1,30 @@
 /datum/keybinding/human
-    category = CATEGORY_HUMAN
-    weight = WEIGHT_MOB
+	category = CATEGORY_HUMAN
+	weight = WEIGHT_MOB
 
+/datum/keybinding/human/down(client/user)
+	return !isnull(SEND_SIGNAL(user.mob, keybind_signal)) //True or False is not important here. If there's a return, we've triggered a proc.
```
Comment (spookydonut) [inline code/datums/keybinding/human.dm]:
```suggestion
	return (SEND_SIGNAL(user.mob, keybind_signal) & COMSIG_KB_ACTIVATED) 
```

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -51,6 +51,9 @@
 
 	randomize_appearance()
 
+	RegisterSignal(src, COMSIG_KB_QUICKEQUIP, .proc/quick_equip)
```
Comment (spookydonut) [inline code/modules/mob/living/carbon/human/human.dm]: add wrapper procs for these and return COMSIG_KB_ACTIVATED

**Comment by Rohesie** [comment]
Suggestion applied.

File: code/datums/keybinding/human.dm [hunk 5]
```diff
@@ -1,39 +1,30 @@
 /datum/keybinding/human
-    category = CATEGORY_HUMAN
-    weight = WEIGHT_MOB
+	category = CATEGORY_HUMAN
+	weight = WEIGHT_MOB
 
+/datum/keybinding/human/down(client/user)
+	return CHECK_BITFIELD(SEND_SIGNAL(user.mob, keybind_signal), COMSIG_KB_ACTIVATED)
```
Comment (spookydonut) [inline code/datums/keybinding/human.dm]: you should check that theres a valid keybind_signal value.

**Pull request merged**